prov/efa/src/efa_prov_info.c

```diff
@@ -384,13 +371,19 @@ int efa_prov_info_alloc(struct fi_info **info,
 	if (!fi)
 		return -FI_ENOMEM;
 
-	fi->caps		= ep_dom->caps;
+	if (ep_type != FI_EP_RDM && ep_type != FI_EP_DGRAM) {
+		EFA_WARN(FI_LOG_DOMAIN, "Unsupported endpoint type: %d\n",
+			 ep_type);
```

Comment:
Should we fail here?

Reply:
Yes, we should. In new revision, I returned `-FI_ENODATA` here.